_posts/2020-03-06-introducing-mirrormaker2.md

```diff
@@ -0,0 +1,247 @@
+Pretty good, right?
+
+This is MirrorMaker unleashed, ready to fulfill the potential we always knew it had.
+
+### Bidirectional opportunities
+
+In the previous version of MirrorMaker, the topic name in the source cluster is automatically created in the downstream cluster.
+Fine, to a point.
+But there is no distinction between _Topic-1.Partition-1_ in the source cluster and _Topic-1.Partition-1_ in the target cluster.
+Essentially, you are limited to _active/passive_ synchonization.
+MirrorMaker 2.0 solves this by introducing the concept of _remote_ topics.
+
+_Remote_ topics are created from the originating cluster by the `MirrorSourceConnector`.
+They are distinguished by automatic renaming that appends the name of cluster to the name of the topic.
+Our _Topic-1.Partition-1_ from the source cluster becomes the never-to-be-confused _Cluster-1-Topic-1.Partition-1_ in the target cluster.
+
+As you can see here, _remote_ topics can be easily identified so that they are not returned to the source cluster.
+
+![Topic renaming with MirrorMaker 2.0](/assets/2020-03-12-mirrormaker-renaming.png)
+
+Why _remote_ topic and not just _source_ topic?
+Because each source cluster can also be a target cluster in a bidirectional configuration.
+This feature also means that it is no longer necessary to aggregate data in a separate cluster.
+
+A rose might be a rose by any other name, but when it comes to replicating topics in MirrorMaker 2.0 we can't afford such ambiguity.
+Particularly if we want to create a bidirectional configuration.
+
+The approach to topic renaming opens up a world of bidirectional opportunities.
+You can now have an _active/active_ cluster configuration that feeds data to and from each cluster.
+For this, you need a MirrorMaker 2.0 cluster at each destination.
+
+### Self-regulating topic replication
+
+Topic replication and consumer group replication is defined using regular expression patterns to _whitelist_ or _blacklist_ topics and consumer groups:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1alpha1
+kind: KafkaMirrorMaker2
+metadata:
+  name: my-mirror-maker2
+  spec:
+    # ...
+    topicsPattern: ".*"
+    groupsPattern: "group1|group2|group3"
+```
+
+You use `topicsBlacklistPattern` and ` groupsBlacklistPattern` if you want to use blacklists.
+
+Pretty similar to before, but previously topic configuration was not replicated.
+The target cluster had to be manually maintained to match topic configuration changes in the source cluster.
+
+This time the topic configuration is now automatically synchronized between source and target clusters through configuration properties.
+Configuration changes are propagated to remote topics so that new topics and partitions are detected and created.
+The need for rebalancing vanishes.
+
+### Offset tracking and mapping
+
+In the old version of MirrorMaker, the offset of the source topic in the target cluster begins when the replication begins.
+The `__consumer-offset topic` is not mirrored.
```

Comment:
The topic name here should be `__consumer_offsets` and I think that perhaps “topic” was meant to be outside the code markup?

Reply:
Yes. You're right. Updated.